lib/bundle/commands/add.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Bundle
+  module Commands
+    module Add
+      module_function
+
+      def run(*args, global: false, file: nil)
+        raise UsageError, "No arguments were specified!" if args.blank?
+
+        type = :brew # default to brew
+        name = args.first # only support one formula at a time for now
+        options = {} # we don't currently support passing options
+
+        # read the relevant Brewfile
+        parsed_entries = Bundle::Dsl.new(Brewfile.read(global: global, file: file)).entries
+
+        # check each of the entries in the specified Brewfile
+        parsed_entries.each do |entry|
+          # raise an error if the entry already exists in the Brewfile
+          # this could also be a noop, or print a friendly message
+          raise RuntimeError if entry.name == name
+        end
+
+        # need some help / pointers here
+        # is it possible to use Bundle::Dsl to create an in memory representation
+        #of the brewfile that we read, add an entry and then dump that back to the file?
```

Comment:
This is where I could use the most help

Reply:
This is the existing dumper code for this: https://github.com/Homebrew/homebrew-bundle/blob/034459da8e6cd11076c578f0ac067fd50d1b14fa/lib/bundle/brew_dumper.rb#L18-L39

It's probably possible to do that but you may have more luck trying to inject your desired formulae state into the existing logic (overriding `@formulae` somehow).